### ISSUE TITLE
PIXI.Graphics initially drawing transparent

### ISSUE THREAD

**Issue by btmorex** [pixijs/pixi.js#5520]
### Expected Behavior

I'm trying to draw a white bar attached to the player sprite.

### Current Behavior

The bar is drawn transparent initially. You can see the area blocked off in the upper right here: https://imgur.com/5gsTd06 Oddly enough, after firing the ship's weapon a few times which essentially creates a few more sprites that quickly disappear the graphics turn white and stay that way. As in: https://imgur.com/gI9ilpF This is strange because I'm completely clearing the stage and recreating everything every frame.

### Steps to Reproduce

Here's my code:

```typescript
const createCapacitorBar = (capacitor: Capacitor) => {
  const bar = new PIXI.Graphics();
  bar.beginFill(0xffffff);
  bar.drawRect(0, 0, 100, 100);
  bar.endFill();
  return bar;
};
```

And elsewhere:

```typescript
    // add player
    const playerSprite = createSprite(player.texture, player.scale, player.static);
    playerSprite.addChild(createCapacitorBar(player.capacitor));
    this.stage.addChild(playerSprite);
    this.stage.addChild(createCollisionShape(player));
```

### Environment

- **`pixi.js` version**: v5.0.0-rc.2
- **Browser & Version**: Chrome 73
- **OS & Version**: Ubuntu 18.04.2

**Comment by btmorex** [comment]
Oh, one more thing, if I use `pixi.js-legacy` and force a canvas renderer the problem goes away. It doesn't appear to be a problem with my graphics card or drivers though because my laptop has both intel and nvidia cards and it happens with both. Also, occurs with both firefox and chrome.

**Comment by ivanpopelyshev** [comment]
I see why it can happen, but it requires other Graphics objects on stage. What other Graphics do you have?

**Comment by btmorex** [comment]
I'm drawing my collision shapes right now, so for example:

```typescript
      const shape = new PIXI.Graphics();
      shape.lineStyle(1, color);
      shape.drawCircle(
        entity.static.position[0],
        entity.static.position[1],
        entity.shape.radius,
      );
      shape.endFill();
      return shape;
```

and

```typescript
      const shape = new PIXI.Graphics();
      shape.lineStyle(1, color);
      shape.drawCircle(entity.static.position[0], entity.static.position[1], 1);
      shape.endFill();
      return shape;
```

**Comment by ivanpopelyshev** [comment]
In 2 minutes this will be available, please test it: http://pixijs.download/dev-graphics-telepathy/pixi-legacy.js 

I can make several attempts or you can give me demo that reproduces the bug.

**Comment by btmorex** [comment]
Is it possible you can send me a diff/patch that I can apply locally? I use npm and pull in pixi as part of a build process.

**Comment by ivanpopelyshev** [comment]
Unfortunately, I cant just make a method that patches pixi from outside in that case, because there's local constant array in the context :(

Im not sure which file should I use for diff. Always trouble with you, tree-shakers!

You can checkout pixijs, switch to `dev-graphics-telepathy` branch, run

```js
npm i
npm run build
cd bundles/pixi.js-legacy
npm link
```

then in your project

```
npm link pixi.js-legacy
```

That way there'll be local link to local pixi :)

**Comment by btmorex** [comment]
No change with that branch and npm link. If it would help, I could give you access to "game" although my deploy process right now is only setup for production builds so it would be minified.

**Comment by btmorex** [comment]
Two other things I noticed that might be interesting:

1.) It appears that the trigger to get them (health bars) drawing correctly is color switching my other use of PIXI.Graphics (collision shapes). The health bars are transparent until something collides temporarily changing the collision shape color from white to red.

2.) My game is multiple player. If I connect with one player and fire a few shots. Open another tab and connect with another player, the new player's health bar instead of being completely transparent appears to have a partial sprite of one of the bullets again until they are "fixed" by a collision. It's almost as if it's being drawn with uninitialized graphics card memory. (is that possible?)

**Comment by ivanpopelyshev** [comment]
email: ivan,popelyshevATgmail.com

That's what i thought, black invisible stuff. or not black, something else. My changes should fix that, but somehow they didnt, hm.

**Comment by btmorex** [comment]
Your fix may be good. I just checked the build output and it's not picking up the fix, I think because it's pulling in the ES version from the linked package. The bundle definitely has the fix, but that's not being included. I'm about to get on a flight so will have to debug further later.

**Comment by ivanpopelyshev** [comment]
Its not. I found problems with it, and after I thought about problem for two more hours, I see several places that can cause problems but I cant reproduce it. Somehow, combination of all if's work fine on my challenges.

**Comment by ivanpopelyshev** [comment]
As a workaround, I recommend to change `graphics.alpha=0` instead of `graphics.beginFill(someColor, 0.0)`

**Comment by btmorex** [comment]
Had some laptop trouble, but finally got around to testing the workaround and it doesn't appear to work. Latest code is like this:

```javascript
  const bar = new PIXI.Graphics();
  bar.alpha = 0.25;
  bar.beginFill(0xffffff);
  bar.drawRect(-15 + ship.static.position[0], -25 + ship.static.position[1], 30, 3);
  bar.endFill();
  bar.alpha = 1;
  bar.beginFill(0xffffff);
  bar.drawRect(
    -15 + ship.static.position[0],
    -25 + ship.static.position[1],
    30 * (ship.combat.capacitor.currentCharge / ship.combat.capacitor.maxCharge),
    3,
  );
  bar.endFill();
  return bar;
```

**Comment by ivanpopelyshev** [comment]
Oh you have transparent and opaque in the same graphics. Strange, I tested it - it does work in current version. I really really need your reduced code.

Alpha affects whole graphics but i thought you have two graphics and not one.

**Comment by btmorex** [comment]
https://gist.github.com/btmorex/cf9335d5d4bbcb80e6d697456e988e3d

**Comment by ivanpopelyshev** [comment]
YEAH

V4: https://www.pixiplayground.com/#/edit/9mpCrPc43Is76lv61Kqqm
V5: https://www.pixiplayground.com/#/edit/1WCb9fBbUQieUK2IdMKzR

**Comment by GoodBoyDigital** [comment]
FYI, This is fixed by @ivanpopelyshev PR here: #5546 👍
